adapters/mexico.js

```diff
@@ -0,0 +1,195 @@
+    const hexCodes = {
+      '&#xF3;': 'ó',
+      '&#xED;': 'í',
+      '&#xE9;': 'é',
+      '&#xE1;': 'á',
+      '&#xC1;': 'Á',
+      '&#xCD;': 'Í',
+      '&#xF1;': 'ñ',
+      '&#xD3;': 'Ó',
+      '&#xC9;': 'É',
+      '&#xFA;': 'ú'
+    };
+    Object.keys(hexCodes).forEach(h => {
+      while (place.search(h) !== -1) {
+        place = place.replace(h, hexCodes[h]);
+      }
+    });
+    const hexPos = place.search('&');
+    if (hexPos !== -1) {
+      console.log(place.substring(hexPos, hexPos + 6));
+    }
+    place = place.split('<br>');
+    const locationMarkers = ['Municipio:', 'Colonia:', 'Estado:'];
+    for (let i of locationMarkers) {
+      var found = false;
+      for (let j in place) {
+        if (place[j].search(i) !== -1) {
+          template['city'] = place[j].replace(i, '').trim();
+          found = true;
+        }
+      }
+      if (found) break;
+    }
+  };
+  var measurements = [];
+  // Loops through each oage
+  pages.forEach(page => {
+    var $ = cheerio.load(page);
+    // base template
+    var template = {
+      attribution: [{name: 'Sinaica.inecc', url: 'https://sinaica.inecc.gob.mx/index.php'}],
+      averagingPeriod: {unit: 'hours', value: 1}
+    };
+    // checks if page has any values to read
+    if ($($('#tabs-1').get(0)).text().trim() !== 'No hay datos disponibles de las últimas 24 horas.') {
+      // finds the city
+      $('.tbl-est.table tr').each((i, e) => {
+        if ($('th', e).text() === 'Dirección postal:') {
+          getCity($('td', e).html().trim(), template);
+        }
+      });
+      // Tries to find location, and values in the document script of the page, and then adds them to measurements
+      try {
+        // base documentscript
+        var values = $('script')
+          .toArray()
+          .map(script => $(script).html())
+          .filter(script => script.search('conts = {') !== -1)[0];
+
+        // Metadata from the documentscript, turns it into json to get latitude, longitude and location
+        var meta = values.substring(values.indexOf('cump = {'));
+        meta = meta.substring(String('cump = {').length - 1, meta.indexOf('};') + 1);
+        meta = JSON.parse(meta);
+        template['coordinates'] = {
+          latitude: Number(meta.lat),
+          longitude: Number(meta.lat)
+        };
+        template['location'] = meta.nom;
+
+        // formats the data and values from the documentscript into readable data
+        var data = values.substring(values.indexOf('conts = {'));
+        data = data.substring(String('conts = {').length - 1, data.indexOf('};') + 1);
+        data = JSON.parse(data);
+        // Loops through all parameters of the site
+        Object.values(data).forEach(param => {
+          // Loops through all the measurements for each parameter
+          param.forEach(d => {
+            if (d != null) {
+              const dateMoment = moment.tz(d.fecha + ' ' + d.hora, 'YYYY-MM-DD H', 'America/Mexico_City');
```

Comment:
It looks like some cities are in a different timezone to Mexico City, for example [Chihuahua ](https://sinaica.inecc.gob.mx/estacion.php?estId=428) is in UCT-8 (or UTC-7 in the summer) while most other cities are UTC-6. Hopefully it should only be a few cities where this is the case but it does list the timezone after `Zona horaria:`